homeassistant/components/discovery.py

```diff
@@ -55,6 +56,7 @@
     SERVICE_DECONZ: ('deconz', None),
     SERVICE_DAIKIN: ('daikin', None),
     SERVICE_SAMSUNG_PRINTER: ('sensor', 'syncthru'),
+    SERVICE_HOMEKIT: ('homekit_controller', None),
```

Comment:
So I want to treat this one with caution. If the next release of Home Assistant all of a sudden starts discovering devices that are already added to Home Assistant via their own API, people will be pissed. Especially because right now there is no way to disable it without disabling all of a specific discovery type.

This will be less of an issue if homekit_controller would use the new config entries support instead of the configurator, as it won't trigger new states in the UI. (video of it in action [here](https://www.youtube.com/watch?v=_qpKe-FW9Yk), more info on config entries [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/config_entries.py#L1), example config flow [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/hue/config_flow.py#L43)). Config entries are currently behind a feature flag (add `config_entries: 1` under `config:`) but will be available for all in the next release.

However, to not blow this PR up any further, instead of converting from using configurator to config entries, you can just change the discovery component to make `SERVICE_HOMEKIT` an opt-in discovery. That way, a future PR can convert homekit controller to config entries.

Reply:
Not quite clear on what you mean by opt-in discovery here - the only configuration that discovery seems to have is to ignore things. Should I be adding additional config support for explicitly enabling devices?

Reply:
I think that would be a good idea. I wouldn't mind to tie it directly to homekit.

```yaml
discovery:
  discover_homekit: true  # default is False
```

(I expect this feature to be removed when someone migrates homekit_controller to config entries)